python/node_classification.py

```diff
@@ -0,0 +1,342 @@
+    CHECKPOINT_FREQ = "checkpoint_freq"
+    TOTAL_NO_EPOCHS = "total_no_epochs"
+
+
+class OtherParams:
+    DEVICE_TYPE = "device_type"
+    PATH_TO_MODELS = "/home/mateo/mage/pytorch_models/model"
+
+
+# all None until set_params are executed
+data: Data = None
+model: InductiveModel = None
+opt = None
+criterion = None
+global_params: mgp.Map
+logged_data: mgp.List = []
+
+DEFINED_INPUT_TYPES = {
+    ModelParams.HIDDEN_FEATURES_SIZE: list,
+    ModelParams.LAYER_TYPE: str,
+    TrainParams.NUM_EPOCHS: int,
+    OptimizerParams.OPTIMIZER: str,
+    OptimizerParams.LEARNING_RATE: float,
+    OptimizerParams.WEIGHT_DECAY: float,
+    DataParams.SPLIT_RATIO: float,
+    MemgraphParams.NODE_FEATURES_PROPERTY: str,
+    MemgraphParams.NODE_ID_PROPERTY: str,
+    OtherParams.DEVICE_TYPE: str,
+    TrainParams.CONSOLE_LOG_FREQ: int,
+    TrainParams.CHECKPOINT_FREQ: int,
+    ModelParams.AGGREGATOR: str,
+    DataParams.METRICS: list
+}
+
+DEFAULT_VALUES = {
+    ModelParams.HIDDEN_FEATURES_SIZE: [16],
+    ModelParams.LAYER_TYPE: "SAGE",
+    TrainParams.NUM_EPOCHS: 100,
+    OptimizerParams.OPTIMIZER: "Adam",
+    OptimizerParams.LEARNING_RATE: 0.1,
+    OptimizerParams.WEIGHT_DECAY: 5e-4,
+    DataParams.SPLIT_RATIO: 0.8,
+    MemgraphParams.NODE_FEATURES_PROPERTY: "features",
+    MemgraphParams.NODE_ID_PROPERTY: "id",
+    MemgraphParams.NODE_CLASS_PROPERTY: "class",
+    OtherParams.DEVICE_TYPE: "cpu",
+    TrainParams.CONSOLE_LOG_FREQ: 5,
+    TrainParams.CHECKPOINT_FREQ: 5,
+    ModelParams.AGGREGATOR: "mean",
+    DataParams.METRICS: ["loss", "accuracy", "f1_score", "precision", "recall", "num_wrong_examples"],
+}
+
+
+##############################
+# set model parameters
+##############################
+
+def declare_globals(params: mgp.Map):
+    """This function declares dictionary of global parameters to given dictionary.
+
+    Args:
+        params (mgp.Map): given dictionary of parameters
+    """
+    global global_params
+    global_params = params
+
+
+def declare_model_and_data(ctx: mgp.ProcCtx):
+    """This function initializes global variables data, model, opt and criterion.
+
+    Args:
+        ctx (mgp.ProcCtx): current context
+    """
+    global global_params
+    nodes = list(iter(ctx.graph.vertices))
+
+    reindexing = {}
+    for i in range(len(nodes)):
+        reindexing[i] = nodes[i].properties.get(global_params["node_id_property"])
+
+    edges = []
+    for vertex in ctx.graph.vertices:
+        for edge in vertex.out_edges:
+            edges.append(edge)
+
+    global data
+    data = convert_data(
+        nodes, edges, global_params[DataParams.SPLIT_RATIO], global_params, reindexing)
+
+    # print(data)
+    # print(data.edge_index.max())
+
+    global_params[ModelParams.IN_CHANNELS] = np.shape(data.x.detach().numpy())[1]
+    print(global_params[ModelParams.IN_CHANNELS])
+    global_params[ModelParams.OUT_CHANNELS] = len(set(data.y.detach().numpy()))
+    print(len(set(data.y.detach().numpy())))
```

Comment:
`global_params[ModelParams.OUT_CHANNELS], global_params[ModelParams.IN_CHANNELS] = np.shape(data.x.detach().numpy())`

Reply:
No, this is not true `ModelParams.OUT_CHANNELS` is number of classes, not number of samples.